packages/ai/core/tool/mcp.ts

```diff
@@ -0,0 +1,209 @@
+import { Client } from '@modelcontextprotocol/sdk/client/index.js';
+import {
+  StdioClientTransport,
+  StdioServerParameters,
+} from '@modelcontextprotocol/sdk/client/stdio.js';
+import {
+  SSEClientTransport,
+  SSEClientTransportOptions,
+} from '@modelcontextprotocol/sdk/client/sse.js';
+import {
+  ListToolsResult,
+  CallToolResultSchema,
+  CallToolResult,
+} from '@modelcontextprotocol/sdk/types';
+import { RequestOptions } from '@modelcontextprotocol/sdk/shared/protocol';
+
+import { jsonSchema, Schema } from '@ai-sdk/ui-utils';
+import { JSONSchema7 } from '@ai-sdk/provider';
+
+import { inferParameters, Tool } from './tool';
+import { z } from 'zod';
+
+type McpStdioServerConfig = StdioServerParameters & {
+  type: 'stdio';
+};
+interface McpSSEServerConfig {
+  type: 'sse';
+  url: string;
+  options?: SSEClientTransportOptions;
+}
+type McpServerConfig = McpStdioServerConfig | McpSSEServerConfig;
+
+interface CreateMcpToolsConfig {
+  servers: Record<string, McpServerConfig>;
+  toolCallOptions?: RequestOptions;
+}
+
+type McpCallToolInputSchema = z.ZodType<any, any, any> | Schema<any>;
```

Comment:
Does the `any` affect type inference down the line (when the mcp tool is used with generateText/streamText)?